src/TLuaInterpreter.cpp

```diff
@@ -10394,13 +10394,660 @@ int TLuaInterpreter::openWebPage(lua_State* L)
+int TLuaInterpreter::setDiscordSmallIconText(lua_State* L)
+{
+    mudlet* pMudlet = mudlet::self();
+    if (!pMudlet->mDiscord.libraryLoaded()) {
+        lua_pushnil(L);
+        lua_pushstring(L, "Discord API is not available");
+        return 2;
+    }
+
+    auto& host = getHostFromLua(L);
+    if (!(host.mDiscordAccessFlags & Host::DiscordLuaAccessEnabled)) {
+        lua_pushnil(L);
+        lua_pushstring(L, "Discord API is disabled in settings for privacy");
+        return 2;
+    }
+
+    if (!pMudlet->mDiscord.discordUserIdMatch(&host)) {
+        lua_pushnil(L);
+        lua_pushstring(L, "Discord API is read-only as you're logged in with a different account in Discord compared to the one you entered for this profile");
+        return 2;
+    }
+
+    if (lua_isstring(L, 1)) {
+        pMudlet->mDiscord.setSmallImageText(&host, QString::fromUtf8(lua_tostring(L, 1)));
+        lua_pushboolean(L, true);
+        return 1;
+    } else {
+        lua_pushfstring(L, "setDiscordSmallIconText: bad argument #%d type (text as string expected, got %s!)", 1, luaL_typename(L, 1));
+        return lua_error(L);
+    }
+}
+
+// TODO: Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#getDiscordSmallIconText
+int TLuaInterpreter::getDiscordSmallIconText(lua_State* L)
+{
+    mudlet* pMudlet = mudlet::self();
+    if (!pMudlet->mDiscord.libraryLoaded()) {
+        lua_pushnil(L);
+        lua_pushstring(L, "Discord API is not available");
+        return 2;
+    }
+
+    auto& host = getHostFromLua(L);
+    if (!(host.mDiscordAccessFlags & Host::DiscordLuaAccessEnabled)) {
+        lua_pushnil(L);
+        lua_pushstring(L, "Discord API is disabled in settings for privacy");
+        return 2;
+    } else if (!(host.mDiscordAccessFlags & Host::DiscordSetSmallIconText)) {
+        lua_pushnil(L);
+        lua_pushstring(L, "Access to Discord small icon text is disabled in settings for privacy");
+        return 2;
+    }
+
+
+    lua_pushfstring(L, pMudlet->mDiscord.getSmallImageText(&host).toUtf8().constData());
+    return 1;
+}
+
+// TODO: Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#setDiscordDetail
+int TLuaInterpreter::setDiscordDetail(lua_State* L)
+{
+    mudlet* pMudlet = mudlet::self();
+    if (!pMudlet->mDiscord.libraryLoaded()) {
+        lua_pushnil(L);
+        lua_pushstring(L, "Discord API is not available");
+        return 2;
+    }
+
+    auto& host = getHostFromLua(L);
+    if (!(host.mDiscordAccessFlags & Host::DiscordLuaAccessEnabled)) {
+        lua_pushnil(L);
+        lua_pushstring(L, "Discord API is disabled in settings for privacy");
+        return 2;
+    }
+
+    if (!pMudlet->mDiscord.discordUserIdMatch(&host)) {
+        lua_pushnil(L);
+        lua_pushstring(L, "Discord API is read-only as you're logged in with a different account in Discord compared to the one you entered for this profile");
+        return 2;
+    }
```

Comment:
The above code for checking if the API is available/the right account/disabled for privacy seems to be repeated in most of these functions. 

Should it maybe be abstracted out instead of repeated everywhere? It would make the code for most of these functions short and concise, and make it easier to change later rather than a massive find/replace that could potentially bork something.

Reply:
I'm not sure there is much to improve upon because those 3 lines would be replaced by 2 lines then - a function call and the return statement

Reply:
it is closer to 20 lines, I just suck at selecting lines in github. should start at 10758.

Reply:
Some:
- Push `nil` and the error string inside the check function, return `true` or `false` about success, do the `return 2` in the outer function
- return the error string or `null` depending on success, do the pushing in the outer function
- return a `std::pair<bool,string>` and use similar logic to lua error returns
- (very ugly and not really something I recommend) use a preprocessor function to insert the block including returns

Reply:
Thanks, `std::pair` is a great idea.

I'm struggling with the syntax for passing a flag to a function though...

Reply:
Why would you need to pass a flag? For the privacy option check (which seems to be the one that changes), one could pass that one as a function argument.

Reply:
Hm, right, an argument type of `int` would do the trick here wouldn't it.

Reply:
I don't understand what you want that argument for, to be honest.

Reply:
Me either, thought it was checking the other flags elsewhere and didn't actually read the flag. Time for a break.

Reply:
Oh, you want to pass `Host::DiscordLuaAccessEnabled` and co? I thought we wanted to get rid of the flags 🤔